src/Controller/IndexController.php

```diff
@@ -87,75 +87,77 @@ public function indexAction(LocalBusinessRepository $repository, CacheInterface
+        $featured = $repository->findFeatured();
 
-                [ $shops, $shopsCount ] =
-                    $this->getItems($repository, $type, $projectCache, $cacheKey, $timingRegistry);
+        if (count($featured) > 0) {
+            $sections[] = [
+                'type' => null,
+                'title' => 'homepage.featured',
+                'shops' => $featured,
+                'type_key_plural' => null,
+                'show_more' => false,
+            ];
+        }
 
-                $sections[] = [
-                    'type' => $type,
-                    'shops' => $shops,
-                    'type_key' => $keyForType,
-                    'type_key_plural' => current($inflector->pluralize($keyForType)),
-                ];
+        $exclusives = $repository->findExclusives();
 
-                if (count($sections) >= 3) {
-                    break;
-                }
-            }
-
-        } else {
+        if (count($exclusives) > 0) {
+            $sections[] = [
+                'type' => null,
+                'title' => 'homepage.exclusives',
+                'shops' => $exclusives,
+                'type_key_plural' => null,
+                'show_more' => false,
+            ];
+        }
 
-            $shops = $repository->withoutTypeFilter()->findAllForType();
+        $news = $repository->findLatest(self::LATEST_SHOPS_LIMIT);
 
-            $iterator = new SortableRestaurantIterator($shops, $timingRegistry);
-            $shops = iterator_to_array($iterator);
+        $sections[] = [
+            'type' => null,
+            'title' => 'homepage.shops.new',
+            'shops' => $news,
+            'type_key_plural' => null,
+            'show_more' => false,
+        ];
 
-            if (count($shops) > 0) {
+        $countByCuisine = $repository->countByCuisine();
 
-                $types = [];
-                foreach ($shops as $shop) {
-                    $types[] = $shop->getType();
-                }
-                $types = array_unique($types);
+        $cuisines = [];
 
-                $keysForTypes = array_map(fn ($t) => LocalBusiness::getKeyForType($t), $types);
+        foreach ($countByCuisine as $cuisine) {
+            if ($cuisine['cnt'] >= self::MIN_SHOPS_PER_CUISINE) {
+                $shopsByCuisine = $repository->findByCuisine($cuisine['id']);
+                $cuisines[] = [
+                    'name' => $cuisine['name'],
+                    'shops' => $shopsByCuisine,
+                    'view_all_path' => 'restaurants_by_cuisine',
+                ];
 
-                if (count($shops) >= (self::ITEMS_PER_ROW * 2)) {
-                    $shops = array_slice($shops, 0, (self::ITEMS_PER_ROW * 2));
-                } elseif (count($shops) >= self::ITEMS_PER_ROW) {
-                    $shops = array_slice($shops, 0, self::ITEMS_PER_ROW);
+                if (count($sections) >= self::MAX_SECTIONS) {
```

Comment:
As cuisine sections are stored in another variable, this makes this `if` useless. Let's manage all sections the same way, and move the logic into the controller.